src/main/webapp/app/upload/upload.component.ts

```diff
@@ -12,12 +19,74 @@ export class UploadComponent {
     error: string;
     uploadResponse = { status: '', message: '' };
 
+    currentPool: Map<String, Pool> = new Map<String, Pool>();
+    newPool: Map<String, Pool> = new Map<String, Pool>();
+
+    public poolType = PoolType;
+
+    currentConflictPool: Pool[] = [];
+    newConflictPool: Pool[] = [];
+
+    merged: Pool[] = [];
+
+    choosePattern = new Map();
+
     public files: UploadFile[] = [];
     file: File;
 
-    constructor(private uploadService: UploadService, private router: Router) {}
+    constructor(private uploadService: UploadService, private router: Router, private poolService: PoolListService) {}
+
+    async onSubmit() {
+        if (!this.file) {
+            return (this.error = 'Please insert file');
+        }
+        this.clearField();
+        const formData = new FormData();
+        formData.append('file', this.file);
+
+        const oldOnesPromise = this.poolService.getPools().toPromise();
+        const newOnesPromise = this.uploadService.parse(formData).toPromise();
+
+        const oldOnes = await oldOnesPromise;
+        const newOnes = await newOnesPromise;
+
+        oldOnes.body.forEach(pool => this.currentPool.set(pool.poolId, pool));
+        newOnes.body.forEach(pool => this.newPool.set(pool.poolId, pool));
+
+        if (this.currentConflictPool.length === 0 && this.newConflictPool.length === 0) {
+            this.getConflicts();
+        }
+    }
+
+    onSave() {
+        this.choosePattern.forEach((value, key) => {
+            let item;
+            if (value === this.poolType.New) {
+                item = this.newPool.get(key);
+            } else {
+                item = this.currentPool.get(key);
+            }
+            this.merged.push(item);
+        });
+        const formData = new FormData();
+        formData.append('pools', JSON.stringify(this.merged));
+        this.uploadService.save(formData).subscribe(
```

Comment:
Let's make alerts look like in method `onForceSave()` please

Reply:
done